pipeline/src/pipeline.rs

```diff
@@ -549,9 +550,32 @@ impl<T: FieldElement> Pipeline<T> {
 
         if self.arguments.export_witness_csv {
             if let Some(path) = self.path_if_should_write(|name| format!("{name}_columns.csv"))? {
-                // TODO: Handle multiple sizes
-                let fixed = get_uniquely_sized_cloned(fixed).unwrap();
-                let columns = fixed.iter().chain(witness.iter()).collect::<Vec<_>>();
+                // get the column size for each namespace. This assumes all witness columns of the same namespace have the same size.
+                let witness_sizes: HashMap<&str, u64> = witness
+                    .iter()
+                    .map(|(name, values)| {
+                        let namespace = name.split("::").next().unwrap();
+                        (namespace, values.len() as u64)
+                    })
+                    .fold(Default::default(), |mut map, (name, size)| {
+                        map.entry(name).or_insert(size);
+                        map
+                    });
```

Comment:
Isn't this just
```suggestion
                    .collect();
```
?